File: app/src/main/java/com/geridea/trentastico/network/request/listener/LessonsDifferenceListener.java
package com.geridea.trentastico.network.request.listener;


/*
 * Created with ♥ by Slava on 31/03/2017.
 */

import com.geridea.trentastico.network.request.LessonsDiffResult;

public interface LessonsDifferenceListener {

    void onLoadingError();

    void onRequestCompleted();

    void onNumberOfRequestToSendKnown(int numRequests);

    void onNoLessonsInCache();

    void onDiffResult(LessonsDiffResult lessonsDiffResult);
}

File: app/src/main/java/com/geridea/trentastico/network/controllers/listener/LessonsDifferenceListener.kt
package com.geridea.trentastico.network.controllers.listener


/*
 * Created with ♥ by Slava on 31/03/2017.
 */

import com.geridea.trentastico.network.request.LessonsDiffResult

interface LessonsDifferenceListener {

    fun onLoadingError()

    fun onRequestCompleted()

    fun onNumberOfRequestToSendKnown(numRequests: Int)

    fun onNoLessonsInCache()

    fun onDiffResult(lessonsDiffResult: LessonsDiffResult)
}
